src/Symfony/Component/Console/Helper/ProgressBar.php

```diff
@@ -519,14 +519,14 @@ private static function initPlaceholderFormatters(): array
                 return Helper::formatTime(time() - $bar->getStartTime());
             },
             'remaining' => function (self $bar) {
-                if (!$bar->getMaxSteps()) {
+                if (!isset($bar->max)) {
```

Comment:
I think this should be reverted, same for below.
Also, on line 53, I think we need to change the definition of `max` to `private int $max = 0;` as `$max` can never be null (see the constructor).

Reply:
I do agree of course that the check will never yield true due to the constructor initialization. Have not spotted that. My bad. However, reverting the lines you mentioned would render the PR void and the issue would still exist.

Thing is, with `$max` always being an integer, there is no semantically correct way to differentiate between a progressbar with a defined maximum and a progressbar without, ie. a progressbar and a throbber, respectively.

Hence, maybe the way to go would be the other way around - keep the definition as is (`?int $max = null`) and introduce checks for `null` where needed?

Will dig into that soon. Thanks in the meanwhile.